### ISSUE TITLE
feature request: add ability to specify location of database file (iOS)

### ISSUE THREAD

**Issue by npomfret** [couchbaselabs/react-native-couchbase-lite#120]
(note to self!)

I'm implementing an app extension which needs access to the database.  Using an App Group it's possible to create a shared area of the file system that both the main app and the app extension can access.
 
Currently the cbl db is written to `[CBLManager defaultDirectory]`, but I need to write to the path returned by `[fileManager containerURLForSecurityApplicationGroupIdentifier:@"group.com.mygroupname"]`

I think maybe changing `initWithAuth` to `init` and passing in a _options_ object that can contain the auth params as well as others (such as the directory) would be a good way to go.

**Comment by jamesnocentini** [comment]
> I think maybe changing initWithAuth to init

Yes that could work in order to provide more options in general. But the directory to create the database is specified when the database is created (through the REST API). The `init` call only starts the listener. How would the options specified in `init` be used when the database is created?

**Comment by npomfret** [comment]
Good question, I'll have a play around and see what I can do.  I've got something working in my fork but it bares little resemblance to master any more.

Another thing I'd like to change (or add) is the default file permissions.  I found that using `NSFileProtectionCompleteUntilFirstUserAuthentication` resulted in much better behaviour when the db was being used while the app was in the background (during a push notification for example).

**Comment by npomfret** [comment]
@jamiltz 

I have this working in my fork (which is very far off master now unfortunately).  It only works for iOS as the android API doesn't seem to give you the option of where to create the file.  I've also changed the return type so there's no need to parse the url in the JS code, you can just pick the values you need.

iOS: https://github.com/npomfret/react-native-couchbase-lite/blob/master/ios/ReactCBLite.m#L27

android: https://github.com/npomfret/react-native-couchbase-lite/blob/master/android/src/main/java/me/fraserxu/rncouchbaselite/ReactCBLite.java#L97

If you think it looks ok I'll make a similar change here.

**Comment by jamesnocentini** [comment]
Oh, that's really nice! I didn't realize the default database directory could be specified on the manager initializer. Yes, if you can submit a PR for this change I'll make sure to test and merge it.
